contracts/Registrar.sol

```diff
@@ -109,31 +102,44 @@ contract Registrar {
 
     constructor(
         ENS _ens,
-        ERC20Burnable _rad,
-        address _admin
+        RadicleToken _rad,
+        address _admin,
+        uint _minCommitmentAge
     ) {
         ens = _ens;
         rad = _rad;
         admin = _admin;
+        minCommitmentAge = _minCommitmentAge;
     }
 
     // --- USER FACING METHODS ---
 
     /// Commit to a future name registration
     function commit(bytes32 commitment) external {
         require(commitments.commited(commitment) == 0, "Registrar::commit: already commited");
-        require(rad.balanceOf(msg.sender) >= fee, "Registrar::register: insufficient rad balance");
 
         rad.burnFrom(msg.sender, fee);
         commitments.commit(commitment);
 
         emit CommitmentMade(commitment, block.number);
     }
 
+    /// Commit to a future name and submit permit in the same transaction
+    function commitWithPermit(bytes32 commitment, address owner, address spender, uint256 value,
+                              uint256 deadline, uint8 v, bytes32 r, bytes32 s) external {
+        rad.permit(owner, spender, value, deadline, v, r, s);
+
+        rad.burnFrom(msg.sender, fee);
+        commitments.commit(commitment);
+
+        emit CommitmentMade(commitment, block.number);
```

Comment:
I think we can use the existing `commit` method here, which also checks to make sure that commitments are not being reused (I think there is a griefing vector where an attacker can continuously resubmit someone elses commitment, ensuring that they can never register their name).

Reply:
although perhaps that attack doesn't work if we have a signature on the commitment.... :thinking:

Reply:
ah, yes, you are right that  `require(commitments.commited(commitment) == 0, "Registrar::commit: already commited"); ` is still missing. 
Notice that we still ask for payment from `msg.sender` here, so changing the commitment would do you no good

Reply:
> Notice that we still ask for payment from msg.sender here, so changing the commitment would do you no good

true!